synapse/storage/databases/main/event_federation.py

```diff
@@ -298,9 +298,13 @@ def _get_auth_chain_ids_using_cover_index_txn(
                     )
 
         # Add the initial set of chains, excluding the sequence corresponding to
-        # initial event.
+        # initial event. Ultimately, the chains dict will be what is pulled from the
+        # database, there is a chance that a sequence number here will end up being a
+        # '0', which doesn't exist. Don't bother sending that to the database query.
         for chain_id, seq_no in event_chains.items():
-            chains[chain_id] = max(seq_no - 1, chains.get(chain_id, 0))
+            max_sequence_result = max(seq_no - 1, chains.get(chain_id, 0))
```

Comment:
Do we understand when max_sequence_result is 0?

I think I've convinced myself that sequence numbers are strictly positive, and chains is a map to sequence numbers. Therefore we would need to have `seq_no == 1 and chain_id not in chains`. Is there any particular meaning to this situation?

Reply:
I guess this corresponds to a brand-new chain that isn't the target of some other chain. Which sounds like the kind of thing that only happens at the start of a room? (cc @erikjohnston: is any of this sane?)

Reply:
This is basically handling the case where an initial event is the first thing in the chain (and nothing else references it). This is actually pretty common, as we start a new chain whenever we see a new type / state key pair.